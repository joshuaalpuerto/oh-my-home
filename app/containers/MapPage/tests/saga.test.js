
import { put } from 'redux-saga/effects'

import mapPageSagas, {
  fetchPlaceSagas,
  fetchPlace
} from '../saga'
import {
  setPlaceAction
} from '../actions'

const args = {
  payload: { placeId: 1 }
}
const results = {
  place: 1
}

describe('MapPage Saga', () => {
  describe('fetchPlaceSagas', () => {
    let fetchPlaceSagasMock

    beforeEach(() => {
      fetchPlaceSagasMock = fetchPlaceSagas()
      const callFetchLatest = fetchPlaceSagasMock.next().value
      expect(callFetchLatest).toMatchSnapshot()
    })

    describe('fetchPlace', () => {
      // pass empty object for params.
      const fetchPlaceMock = fetchPlace(args)

      it('should match the snapshot for setting token', () => {
        const callingAPI = fetchPlaceMock.next().value
        expect(callingAPI).toMatchSnapshot()
      })

      it('should set setPlaceAction', () => {
        // eslint-disable-next-line no-unused-vars
        const settingData = fetchPlaceMock.next(results).value
        expect(settingData).toEqual(
          put(setPlaceAction(results))
        )
      })
    })
  })

  describe('defaultGenerators', () => {
    const generator = mapPageSagas()

    it('should have 1 functions', () => {
      const watchers = generator.next().value
      expect(watchers.length).toEqual(1)
    })
  })
})
