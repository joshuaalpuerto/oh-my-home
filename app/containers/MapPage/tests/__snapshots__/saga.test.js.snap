// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapPage Saga fetchPlaceSagas fetchPlace should match the snapshot for setting token 1`] = `
Object {
  "@@redux-saga/HELPER": true,
  "name": "takeLatest(app/MapPage/GET_PLACE, fetchPlace)",
  "next": [Function],
  "return": [Function],
  "throw": [Function],
  Symbol(Symbol.iterator): [Function],
}
`;

exports[`MapPage Saga fetchPlaceSagas fetchPlace should match the snapshot for setting token 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://qa.omh.sg/test/places?place_id=1",
      Object {
        "method": "GET",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`MapPage Saga fetchPlaceSagas fetchPlace should set setPlaceAction 1`] = `
Object {
  "@@redux-saga/HELPER": true,
  "name": "takeLatest(app/MapPage/GET_PLACE, fetchPlace)",
  "next": [Function],
  "return": [Function],
  "throw": [Function],
  Symbol(Symbol.iterator): [Function],
}
`;
